data_structures/ringbuffer.c

```diff
@@ -0,0 +1,129 @@
+/**
+ * Speedster3030, 2025
+ * Ring Buffer/Circular Queue structure implementation.
+ */
+
+#include<stdio.h>
+#include<stdlib.h>
+
+//ring Buffer/Circular Queue struct
+typedef struct
+{
+        int* arr;
+        int count;
+        int head;
+        int tail;
+        int max;
+}ringBuffer;
+
+//Initialize a new ring buffer
+ringBuffer* new_ringBuffer(int size)
+{
+        ringBuffer* rb=malloc(sizeof(ringBuffer));
+
+        rb->max=size;
+        rb->arr=malloc(size*sizeof(int));
+        rb->count=0;
+        rb->head=size-1;
+        rb->tail=size-1;
+
+        return rb;
+}
+
+//free the structure from memory
+void delete_Buffer(ringBuffer **rb)
+{
+        free((*rb)->arr);
+        free(*rb);
+        *rb=NULL;
+}
+
+//Put a value in the buffer
+int enqueue(ringBuffer *rb,int n)
+{
+        if(rb->count==rb->max)
+        {
+                return -1;
+        }
+
+        rb->arr[rb->tail]=n;
+        rb->tail--;
+        if(rb->tail==-1)
+        {
+                rb->tail=rb->max-1;
+        }
+        rb->count++;
+        return 0;
+}
+
+//remove a value from the buffer
+int dequeue(ringBuffer *rb,int *result)
+{
+        if(rb->count==0)
+        {
+                return -1;
+        }
+
+        int t=rb->arr[rb->head];
+        rb->arr[rb->head]=0;
+        rb->head--;
+        rb->count--;
+        if(rb->head==-1)
+        {
+                rb->head=rb->max-1;
+        }
+
+        *result=t;
+        return 0;
+}
+
+//get the value in front of the queue
+int front(ringBuffer *rb,int *result)
+{
+        if(rb->count==0)
+        {
+                return -1;
+        }
+
+        *result=rb->arr[rb->head];
+        return 0;
+}
+
```

Comment:
You already provide a front() function, but no equivalent rear() function. Since the buffer tracks both ends, it would be useful (and consistent with [Circular Queue implementations](https://www.geeksforgeeks.org/dsa/introduction-to-circular-queue/)) to add a rear() to peek at the last enqueued element.
```suggestion

// get the value of the rear of queue
int rear(ringBuffer *rb, int *result)
{
    if (rb->count == 0)
        return -1;

    int idx = (rb->tail + 1) % rb->max; 
    return 0;
}

```